core/datatrak.c

```diff
@@ -204,7 +204,7 @@ unsigned char *dt_dive_parser(unsigned char *runner, struct dive *dt_dive)
 	ds = get_dive_site_by_name(buffer);
 	dt_dive->dive_site_uuid = ds ? ds->uuid : 0;
 	if (dt_dive->dive_site_uuid == 0)
-		dt_dive->dive_site_uuid = create_dive_site(buffer, dt_dive->when);
+		dt_dive->dive_site_uuid = create_dive_site(buffer, dt_dive->when)->uuid;
```

Comment:
I was gonna say "but what about a NULL pointer?" - and then I read the code of `alloc_or_get_dive_site()` and the fact that we simply do an exit(1) if we run out of memory. Oh. OK. That makes sure there's never a NULL pointer returned, I guess... 😱

Reply:
Moreover note that the dereference is removed in a later commit anyway.